packages/grafana-prometheus/src/components/monaco-query-field/MonacoQueryField.tsx

```diff
@@ -200,23 +200,6 @@ const MonacoQueryField = (props: Props) => {
           editor.onDidContentSizeChange(updateElementHeight);
           updateElementHeight();
 
-          // Whenever the editor changes, lets save the last value so the next query for this editor will be up-to-date.
-          // This change is being introduced to fix a bug where you can submit a query via shift+enter:
-          // If you clicked into another field and haven't un-blurred the active field,
-          // then the query that is run will be stale, as the reference is only updated
-          // with the value of the last blurred input.
-          // This can run quite slowly, so we're debouncing this which should accomplish two things
-          // 1. Should prevent this function from blocking the current call stack by pushing into the web API callback queue
-          // 2. Should prevent a bunch of duplicates of this function being called as the user is typing
-          const updateCurrentEditorValue = debounce(() => {
-            const editorValue = editor.getValue();
-            onChangeRef.current(editorValue);
-          }, lpRef.current.datasource.getDebounceTimeInMilliseconds());
-
-          editor.getModel()?.onDidChangeContent(() => {
-            updateCurrentEditorValue();
-          });
-
           // handle: shift + enter
```

Comment:
Could you share why is it okay to remove this code?

Reply:
It was implemented here https://github.com/grafana/grafana/pull/55513
Then it was altered a bit with this PR https://github.com/grafana/grafana/pull/60191/files
The initial implementation was causing the jump so I removed it.